packages/tables/src/Table/Concerns/CanSummarizeRecords.php

```diff
@@ -7,8 +7,42 @@
 
 trait CanSummarizeRecords
 {
+    protected bool | Closure | null $hasPageSummary = null;
+
+    protected bool | Closure | null $hasTotalSummary = null;
+
+    public function summaries(bool | Closure | null $pageSummary = null, bool | Closure | null $totalSummary = null): static
+    {
+        if ($pageSummary !== null) {
+            $this->hasPageSummary = $pageSummary;
+        }
+
+        if ($totalSummary !== null) {
+            $this->hasTotalSummary = $totalSummary;
+        }
+
+        return $this;
+    }
+
+    public function hasPageSummary(): bool
+    {
+        return (bool) $this->evaluate($this->hasPageSummary ?? true);
+    }
+
+    public function hasTotalSummary(): bool
+    {
+        return (bool) $this->evaluate($this->hasTotalSummary ?? true);
+    }
+
     public function hasSummary(Builder | Closure | null $query): bool
     {
+        if (
+            $this->hasPageSummary === false &&
```

Comment:
What is this condition for? If we need to call `hasPageSummary` and `hasTotalSummary` in the view anyway, do we really need to check here too?

Reply:
if we pass the parameters through the attributes like you said, these won't be necessary anymore. which we have now.